utils/epinow2/functions.py

```diff
@@ -0,0 +1,183 @@
+from datetime import date, datetime, timedelta, timezone
+from uuid import UUID, uuid1
+
+from utils.epinow2.constants import (
+    all_diseases,
+    all_states,
+    nssp_states_omit,
+    shared_params,
+)
+
+
+def generate_timestamp() -> int:
+    """Generates a timestamp of the current time using UTC timezone."""
+    return int(datetime.timestamp(datetime.now(timezone.utc)))
+
+
+def get_reference_date_range(report_date: date) -> tuple[date, date]:
+    """Returns a tuple of the minimum and maximum reference dates
+    based on the report date, in the case that no reference_dates
+    are provided.
+    Parameters:
+        report_date: date of model run
+    """
+    # Convert user-provided report_date to date object
+    if isinstance(report_date, str):
+        report_date = date.fromisoformat(report_date)
+
+    max_report_date = report_date - timedelta(days=1)
+    min_report_date = report_date - timedelta(weeks=8)
+
+    return min_report_date, max_report_date
+
+
+def generate_uuid() -> UUID:
+    """Generates a UUID1 object to associate
+    with job IDs. UUID1 can be sorted by
+    timestamp by default, which is desirable
+    when managing multiple configurations."""
+    return uuid1()
+
+
+def generate_job_id(job_id: UUID | None = None, as_of_date: int | None = None) -> str:
+    """Generate a human-readable slug based on job UUID and as_of_date.
+    Parameters:
+        job_uuid: UUID for job
+        as_of_date: timestamp of model run
+    """
+    job_name = f"Rt-estimation-{job_id.hex}-{datetime.fromtimestamp(as_of_date).isoformat()}".replace(
```

Comment:
Generating job name based on this comment from @natemcintosh https://github.com/CDCgov/cfa-epinow2-pipeline/pull/7#discussion_r1713916309